sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Custom/FastProvisioning.cs

```diff
@@ -0,0 +1,128 @@
+                    var cachedServerNameResponse = await GetCachedServerNameRestClient.ExecuteAsync(Id.SubscriptionId, Id.ResourceGroupName, data.Location, cachedServerNameRequest, cancellationToken).ConfigureAwait(false);
+                    serverName = cachedServerNameResponse.Value.Name;
+                }
+                catch (Exception e)
+                {
+                    cachedServerNameScope.Failed(e);
+                    throw;
+                }
+
+                using var serverScope = _postgreSqlFlexibleServerServersClientDiagnostics.CreateScope("PostgreSqlFlexibleServerCollection.CreateOrUpdate");
+                serverScope.Start();
+                try
+                {
+                    var serverResponse = await _postgreSqlFlexibleServerServersRestClient.CreateAsync(Id.SubscriptionId, Id.ResourceGroupName, serverName, data, cancellationToken).ConfigureAwait(false);
+                    var serverOperation = new FlexibleServersArmOperation<PostgreSqlFlexibleServerResource>(new PostgreSqlFlexibleServerOperationSource(Client), _postgreSqlFlexibleServerServersClientDiagnostics, Pipeline, _postgreSqlFlexibleServerServersRestClient.CreateCreateRequest(Id.SubscriptionId, Id.ResourceGroupName, serverName, data).Request, serverResponse, OperationFinalStateVia.Location);
+                    if (waitUntil == WaitUntil.Completed)
+                        await serverOperation.WaitForCompletionAsync(cancellationToken).ConfigureAwait(false);
+                    return serverOperation;
+                }
+                catch (Exception e)
+                {
+                    serverScope.Failed(e);
+                    throw;
+                }
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Creates a new server using fast provisioning method.
+        /// </summary>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="data"> The required parameters for creating a server. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual ArmOperation<PostgreSqlFlexibleServerResource> FastProvision(WaitUntil waitUntil, PostgreSqlFlexibleServerData data, CancellationToken cancellationToken = default)
```

Comment:
What will be the example of using this method like? Can we possibly add a sample of this in the sample repo if possible?

Reply:
Sure, after this gets merged, I'll add a sample.